feat: Update Code Scanner Docs (#653)
appsmithorg/appsmith-docs: website/docs/reference/widgets/code-scanner.md

## Usage

> QR/Barcodes have taken off because they offer a clear and fast return on investment. It eliminates manual entry of product information at receiving, meaning there are far fewer opportunities for error. You can use a code scanner in some of the below scenarios:

Reviewer: ```suggestion
QR/Barcodes have taken off because they offer a clear and fast return on investment. It eliminates manual entry of product information at receiving, meaning there are far fewer opportunities for error. You can use a code scanner in the below scenarios:
```

> QR/Barcodes have taken off because they offer a clear and fast return on investment. It eliminates manual entry of product information at receiving, meaning there are far fewer opportunities for error. Lets look at some scenarios where a code scanner can be helpful:

Reviewer: ```suggestion
QR/Barcodes have taken off because they offer a clear and fast return on investment. It eliminates manual entry of product information at receiving, meaning there are far fewer opportunities for error. You can use a code scanner in some of the below scenarios:
```

* **Warehouse**- Companies need to know not just their current inventory levels, but the location of those products, always. With the help of a code scanner widget, employees can keep track of their warehouse inventories. They can use a code scanner in their app to scan codes rather than manually entering them.
* **Asset Management**- The most widely used system by businesses for tracking equipment, vehicles, and computers to support their everyday operations is the usage of barcodes. With the code scanner widget, you can keep track of equipment and the location of the assets and keep any pertinent maintenance and repair records.
* **Logistics and Transportation industries**- Effective track and trace methods offer logistics teams the ability to locate, track, and identify objects in real-time, providing better business insight and quicker problem-solving. QR-based tracking eliminates paperwork, reduces errors, and increases productivity.

### Supported QR/Barcode formats

> Appsmith supports the following formats for QR and barcodes:

Reviewer: ```suggestion
The following formats for QR and barcodes are supported:
```

| [Text](code-scanner.md#text)                     | Formatting           | Sets the label text of the widget.                                                                                                                                        | NA                           |
| [Visible](./README.md#visible)                   | Binding & Formatting | Controls widget's visibility on the page. When turned off, the widget won't be visible when the app is published. Learn more about [Visible Property](./README.md#visible | `{{CodeScanner.isVisible}}`  |
| [Disabled](./README.md#disabled)                 | Binding & Formatting | Disables input to the widget. The widget will remain visible to the user but a user input won't be allowed. Learn more about [Disable Property](./README.md#disabled)     | `{{CodeScanner.isDisabled}}` |
| [Animate Loading](./README.md)                   | Formatting           | Controls the loading of the widget.                                                                                                                                       | NA                           |

> | [Tooltip](./README.md#tooltip)                   | Formatting           | It sets a tooltip for the widget. You can add hints or extra information about the required input from the user. Learn more about [Tooltip](./README.md#tooltip)          | NA                           |

Reviewer 2: Where does it show up on the widget?

Author: Tooltip is visible only when Scanner Layout is set to "Click to Scan"

<img width="616" alt="CleanShot 2022-12-08 at 14 52 13@2x" src="https://user-images.githubusercontent.com/22471214/206408146-6300a61d-0373-4074-85c1-3c67fb7e73e9.png">

Moreover, we hide the tooltip property in the property pan when the scanner layout is always on.

<img width="272" alt="image" src="https://user-images.githubusercontent.com/22471214/206408539-7239e0b6-6539-47db-a7f4-d1c082e48e66.png">

## What's next

The following resources may come handy as you need to learn new tricks:

* [Core Concepts](/core-concepts/connecting-to-data-sources/)
* [Appsmith Framework](/reference/appsmith-framework/)
* [JavaScript Editor](/core-concepts/writing-code/javascript-editor-beta/)

## Troubleshooting

> If you encounter any errors during this process, check out this guide on [debugging deployment errors](/help-and-support/troubleshooting-guide/deployment-errors.md). If you are still facing any issues, please reach out to support@appsmith.com or join this [Discord Server](https://discord.com/invite/rBTTVJp) to speak to the Appsmith team directly.

Reviewer: ```suggestion
If you encounter any errors during this process, check out this guide on [debugging deployment errors](/help-and-support/troubleshooting-guide/deployment-errors.md). If you are still facing any issues, please get in touch with support@appsmith.com or raise your query on [Discord Server](https://discord.com/invite/rBTTVJp).
```
